clients/src/main/java/org/apache/kafka/common/config/ConfigResource.java

```diff
@@ -33,7 +33,12 @@ public final class ConfigResource {
      * Type of resource.
      */
     public enum Type {
-        CLIENT_METRICS((byte) 16), BROKER_LOGGER((byte) 8), BROKER((byte) 4), TOPIC((byte) 2), UNKNOWN((byte) 0);
+        GROUP((byte) 32),
```

Comment:
Can we have GROUP(3) instead of GROUP(32) so that it matches with ResourceType used by ACLs. Ideally and according to KIP-133 Config and ACL APIs should have same ResourceType. This is the way librdkafka has adopted the ResourceType. We already have GROUP as `3` in ResourceType and both Config and ACL APIs use the same resource type enum. We cannot create a new Enum right away as these are part of the public API.

Reply:
It seems a little unfortunate that `o.a.k.common.resource.ResourceType` (as used by ACLs) and `o.a.k.common.config.ConfigResource.Type` (as used by config resources) are separate enums. They do not have perfect alignment. For example, `ResourceType.CLUSTER` (the whole cluster) and `ConfigResource.Type.BROKER`(a single broker) are both 4. There are concepts in one of the enums which do not exist in the other, and I expect there's an element of luck to whether values align between them.

I personally don't see any reason why `ConfigResource.Type.GROUP` could not be 3 (which I think would need a tweak to the KIP) which would then align with `ResourceType.GROUP`.

More broadly, they are not the same enums and if we want to keep them aligned as much as possible moving forwards, we need to take steps in the code (such as comments) to ensure that when updates are being made to one of the enums, it's done with awareness of the values used in the other.

@pranavrth I think you're going to have to bite the bullet in librdkafka and define a second enum, when you are able to change the public API.

Reply:
Done.